src/Features/CSharp/Portable/CodeRefactorings/SyncNamespace/CSharpChangeNamespaceService.cs

```diff
@@ -135,89 +192,163 @@ protected override SyntaxList<MemberDeclarationSyntax> GetMemberDeclarationsInCo
         }
 
         /// <summary>
-        /// Try to change the namespace declaration based on the refacoring rules:
-        ///     - if neither declared and target namespace are "" (i.e. global namespace),
+        /// Try to change the namespace declaration based on the following rules:
+        ///     - if neither declared nor target namespace are "" (i.e. global namespace),
         ///     then we try to change the name of the namespace.
         ///     - if declared namespace is "", then we try to move all types declared 
         ///     in global namespace in the document into a new namespace declaration.
         ///     - if target namespace is "", then we try to move all members in declared 
         ///     namespace to global namespace (i.e. remove the namespace declaration).    
         /// </summary>
         protected override CompilationUnitSyntax ChangeNamespaceDeclaration(
-            CompilationUnitSyntax compilationUnit, 
+            CompilationUnitSyntax root, 
             ImmutableArray<string> declaredNamespaceParts, 
             ImmutableArray<string> targetNamespaceParts)
         {
             Debug.Assert(!declaredNamespaceParts.IsDefault && !targetNamespaceParts.IsDefault);
+            var container = root.GetAnnotatedNodes(ContainerAnnotation[0]).Single();
```

Comment:
this looks weird.  What is ContainerANnotation[0]?

Reply:
Also... it's indexable... but sounds singular...

Reply:
hmm, for some reason I thought the method removes annotation only takes an array... fixed.